src/pudl/extract/ferc.py

```diff
@@ -0,0 +1,33 @@
+"""Shared utilities for ferc extraction process."""
+
+import sqlalchemy as sa
+
+
+def add_key_constraints(
+    meta: sa.MetaData, pk_table: str, column: str, pk_column: str | None = None
+) -> sa.MetaData:
+    """Adds primary and foreign key to tables present in meta.
+
+    Args:
+        meta: constraints will be applied to this metadata instance
+        pk_table: name of the table that contains primary-key
+        column: foreign key column name. Tables that contain this column will
```

Comment:
In general these could be composite keys composed of multiple columns (e.g. `report_year` and `respondent_id`). Thus far the only keys we've been able to make work are the simple `respondent_id` and even that has required some repair of its home table.

If it would be easy to let these be composite keys rather than single columns that would be good. If it's not easy, then we should at least note that only simple keys are possible and why that's been okay for the FERC DBF data so far.